src/main/java/com/google/devtools/build/lib/actions/ActionEnvironment.java

```diff
@@ -186,6 +186,23 @@ public ActionEnvironment addFixedVariables(Map<String, String> vars) {
     return new ActionEnvironment(new CompoundEnvironmentVariables(vars, fixedEnv), inheritedEnv);
   }
 
+  /**
+   * Returns a copy of the environment with the given fixed and inherited variables added to it,
+   * <em>overwriting any existing occurrences of those variables</em>.
+   */
+  public ActionEnvironment addVariables(Map<String, String> vars, Set<String> inheritedVars) {
+    if (inheritedVars.isEmpty()) {
+      return addFixedVariables(vars);
+    } else {
+      // TODO: inheritedEnv is currently not optimized for allocation avoidance in the same way as
+      //  fixedEnv.
```

Comment:
Do you plan to address this? It might become an issue.

Reply:
I will look into doing it now, but could also reserve it for a follow-up PR. Let me know what you prefer.

Reply:
I pushed another that implements the optimization.